searchlib/src/vespa/searchlib/common/allocatedbitvector.cpp

```diff
@@ -17,6 +17,21 @@ size_t computeCapacity(size_t capacity, size_t allocatedBytes) {
     return possibleCapacity;
 }
 
+// This is to ensure that we only read size and capacity once during copy
+// to ensure that they do not change unexpectedly under our feet due to resizing in different thread.
+std::pair<BitVector::Index, BitVector::Index>
+extract_size_size(const BitVector & bv) {
+    BitVector::Index size = bv.size();
+    return std::pair<BitVector::Index, BitVector::Index>(size, size);
+}
+
+std::pair<BitVector::Index, BitVector::Index>
+extract_size_capacity(const AllocatedBitVector & bv) {
+    BitVector::Index size = bv.size();
+    BitVector::Index capacity = bv.capacity();
+    return std::pair<BitVector::Index, BitVector::Index>(size, capacity);
```

Comment:
If size and capacity is reduced at the same time on the underlying bitvector then capacity can become less than size here.

Reply:
capacity is updated before size when growing.
However upon shrink that will not be safe.
So I guess a loop and a comment is needed here.

Reply:
Fixed, @toregge PTAL